File: AtCoder/Browns/python/ARC115-C-N_Coloring.py
N = int(input())

A = [0] * (N + 1)
A[1] = 1


def divisors(n):
    res = []
    for i in range(1, int(n**0.5) + 1):
        if n % i == 0:
            res.append(i)
            if n // i not in (i, n):
                res.append(n // i)
    return res


for i in range(2, N + 1):
    divs = divisors(i)
    st = set()
    for d in divs:
        st.add(A[d])
    A[i] = len(st) + 1
print(' '.join(list(map(str, A[1:]))))

File: AtCoder/Browns/cpp/ARC115-C-N_Coloring.cpp
#include <bits/stdc++.h>
#define MAX_N 100'000
using namespace std;
typedef long long ll;

ll N;
ll A[MAX_N + 1];

vector<ll> divisors(ll n) {
  vector<ll> res;
  for (ll i = 1; i * i <= n; ++i) {
    if (n % i == 0) {
      res.push_back(i);
      if (n / i != i && n / i != n) res.push_back(n / i);
    }
  }
  return res;
}

int main() {
  cin >> N;
  A[1] = 1;
  for (ll i = 2; i <= N; ++i) {
    vector<ll> divs = divisors(i);
    set<ll> st;
    for (auto d : divs)
      st.insert(A[d]);
    A[i] = st.size() + 1;
  }
  for (ll i = 1; i <= N; ++i)
    printf("%lld%c", A[i], i == N ? '\n' : ' ');
}